// Copyright (C) 2021 Bloomberg LP
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//  <http://www.apache.org/licenses/LICENSE-2.0>
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package buildgrid.v2;

import "google/api/annotations.proto";
import "google/devtools/build/v1/build_events.proto";
import "google/devtools/build/v1/publish_build_event.proto";

service QueryBuildEvents {
  rpc QueryEventStreams(QueryEventStreamsRequest) 
      returns (QueryEventStreamsResponse) {
    option (google.api.http) = { get: "/v2/{name=event_streams}" };
  }
}

message QueryEventStreamsRequest {
  string build_id_pattern = 1;
}

message QueryEventStreamsResponse {
  message BuildEventStream {
    google.devtools.build.v1.StreamId stream_id = 1;
    repeated google.devtools.build.v1.OrderedBuildEvent events = 2;
  }

  repeated BuildEventStream streams = 1;
}
